spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1736,12 +1737,18 @@ def process_code_analysis(self, check_results):
         self.classfuncdropdown.update()
 
     def show_code_analysis_results(self, line_number, block_data):
-        """Show warning/error messages"""
+        """Show warning/error messages."""
         code_analysis = block_data.code_analysis
-        msglist = ['{0} [{1}]: {2}'.format(
-            src, code, msg) for src, code, _sev, msg in code_analysis]
-        self.show_calltip(_("Code analysis"), msglist,
-                          color='#129625', at_line=line_number)
+        template = '{0} [{1}]: {2}'
+        msglist = [template.format(src, code, msg) for src, code, _sev, msg
+                   in code_analysis]
+
+        self.show_tooltip(
+            _("Code analysis"),
+            msglist,
+            color='#129625',
+            at_line=line_number,
+        )

```

Comment:
When are these warnings hidden? The idea is that they should be hidden without user interaction. Or in other words, they should behave like a regular tooltip (i.e hidden after the mouse is out of the widget).

Reply:
Since the signature call is now much more complex, to keep things more organized I created 2 separate methods. 

`show_tooltip` and `show_calltip`. The names explain what they are using and this allows to move all the styling logic inside the mixin.